### ISSUE TITLE
[Bug]: Combobox value/searchTerm isn't fully reactive

### ISSUE THREAD

**Issue by toxin20** [radix-vue/radix-vue#433]
### Environment

```bash
Radix Vue version: 0.4.1
Nuxt version: 3.7.4
CSS framework: @nuxtjs/tailwindcss@6.8.0
```

### Link to minimal reproduction

https://codesandbox.io/p/sandbox/vue-radix-combobox-test-pfynd8?file=%2Fapp.vue%3A3%2C37

### Steps to reproduce

- You start out with "Apple" selected in the combobox
- Click the button to set the v ref value to "Banana"
- The combobox searchterm is still "Apple". When you open the Combobox you can see that "Banana" is selected though
- Click the button to set the v ref value null
- Again the searchterm stays, even though the selection is nulled

### Describe the bug

- When the v ref from `v-model` is changed from the outside, the selected option updates but the search term text still reflects the old selection. 
- I've set `:search-term=searchTerm` to bind this variable. But as you can see in the CodeSandbox and also in the animated GIF, this variable seems to be always empty

### Expected behavior

- When the `v-model` variable is changed to another value from the outside, the combobox should be set to that option if possible and the search term text should also reflect the newly selected option instead of staying the same. 
- The same goes for setting the `v-model` variable to null, which should clear the combobox selection AND search term text.
- The search term is supposed to be exposed via `:search-term` but it stays empty all the time and does not change. Might need clearing up in the docs on how to use this `:search-term` binding and what for. 

### Context & Screenshots (if applicable)

![CleanShot 2023-10-01 at 10 06 42](https://github.com/radix-vue/radix-vue/assets/123776/b221b80b-eae9-439a-b527-aadace0d3f1f)

**Comment by zernonia** [comment]
@toxin20 could you help me check if [v1.0.0-beta.2](https://github.com/radix-vue/radix-vue/releases/tag/v1.0.0-beta.2) fix this issue?😁

**Comment by toxin20** [comment]
Thanks for taking care of this @zernonia! 

I just upgraded my codesandbox to and tried your new version:
https://codesandbox.io/p/sandbox/vue-radix-combobox-test-pfynd8?file=%2Fpackage.json%3A19%2C2

I couldn't see any change in behavior though, it's still the same:

![CleanShot 2023-10-21 at 16 34 05](https://github.com/radix-vue/radix-vue/assets/123776/7ddcd4ae-15e0-4d11-8c31-b7dd8df0e191)

**Comment by zernonia** [comment]
Added a fix to on latest `1.0.0-beta.3`

Let me know if I'm missing out anything ya!
https://codesandbox.io/p/sandbox/vue-radix-combobox-test-pfynd8?file=%2Fpackage.json%3A19%2C2
